src/System.Management.Automation/engine/remoting/common/RunspaceConnectionInfo.cs

```diff
@@ -3443,7 +3443,26 @@ private void GetContainerPropertiesInternal()
                     //
                     if (RuntimeId == Guid.Empty)
                     {
-                        ContainerObRoot = (string)computeSystemPropertiesType.GetProperty("ObRoot").GetValue(computeSystemPropertiesHandle);
+                        // Since Hyper-V changed this from a property to a field, we can optimize for newest Windows to see if it's a field,
+                        // otherwise we fall back to old code to be compatible with older versions of Windows
+                        var obRootFieldInfo = computeSystemPropertiesType.GetField("ObRoot");
+                        if (obRootFieldInfo != null)
+                        {
+                            ContainerObRoot = obRootFieldInfo.GetValue(computeSystemPropertiesHandle) as string;
+                        }
+                        else
+                        {
+                            var obRootPropertyInfo = computeSystemPropertiesType.GetProperty("ObRoot");
+                            if (obRootPropertyInfo != null)
+                            {
+                                ContainerObRoot = obRootPropertyInfo.GetValue(computeSystemPropertiesHandle) as string;
+                            }
+                        }
```

Comment:
```suggestion
                        ContainerObRoot = (computeSystemPropertiesType.GetField("ObRoot")?.GetValue(computeSystemPropertiesHandle)
                            ?? computerSystemPropertiesType.GetProperty("ObRoot")?.Value(computeSystemPropertiesHandle)) as string;
```

Reply:
As much as I usually like to compress code like this, I have to disagree on this one. I intentionally used these `if`s and `var`s to keep my changes in line with the comments and the existing code above - for better readability.